logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/auth/LogoutUseCase.kt

```diff
@@ -62,23 +62,13 @@ internal class LogoutUseCaseImpl @Suppress("LongParameterList") constructor(
         logoutRepository.onLogout(reason)
 
         when (reason) {
-            LogoutReason.SELF_HARD_LOGOUT -> {
+            LogoutReason.SELF_HARD_LOGOUT, LogoutReason.REMOVED_CLIENT, LogoutReason.DELETED_ACCOUNT -> {
```

Comment:
this will delete user data when a client get remove, like database message history

Reply:
Yup, is that wrong?

Reply:
1. other clients do not have the same behavior.
2. in some cases, clients can miss this event if they are offline for a while.
3. we need to look into what caused the issue mentioned above because this logic was working just fine.

Reply:
lets take it 1:1 and dive deep to get what exactly is going causing the issue

Reply:
Added changed discussed with @MohamadJaara